src/main/java/gyro/google/compute/AbstractAddressResource.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    /**
+     * Type of address to reserve. Valid values are ``INTERNAL`` or ``EXTERNAL``. Defaults to ``EXTERNAL``.
+     */
+    @ValidStrings({ "EXTERNAL", "INTERNAL" })
+    public String getAddressType() {
+        return addressType;
+    }
+
+    public void setAddressType(String addressType) {
+        this.addressType = addressType;
+    }
+
+    /**
+     * Purpose for this resource. Valid values are ``GCE_ENDPOINT``, ``DNS_RESOLVER``, ``VPC_PEERING`` or ``NAT_AUTO``.
+     */
+    @ValidStrings({ "GCE_ENDPOINT", "DNS_RESOLVER", "VPC_PEERING", "NAT_AUTO" })
+    public String getPurpose() {
+        return purpose;
+    }
+
+    public void setPurpose(String purpose) {
+        this.purpose = purpose;
+    }
+
+    /**
+     * URL of the subnetwork in which to reserve the address. If an IP address is specified, it must be wihin the subnetwork's IP range. This field can only be used with ``INTERNAL`` type with a ``GCE_ENDPOINT`` or ``DNS_RESOLVER`` purpose.
+     */
+    public SubnetworkResource getSubnetwork() {
+        return subnetwork;
+    }
+
+    public void setSubnetwork(SubnetworkResource subnetwork) {
+        this.subnetwork = subnetwork;
+    }
+
+    /**
+     * The URL of the network in which to reserve the address. This field can only be used with ``INTERNAL`` type with the ``VPC_PEERING`` purpose.
+     */
+    public NetworkResource getNetwork() {
+        return network;
+    }
+
+    public void setNetwork(NetworkResource network) {
+        this.network = network;
+    }
+
+    /**
+     * The status of the address. Values are ``RESERVING``, ``RESERVED`` or ``IN_USE``.
+     */
+    @Output
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+
+    /**
+     * GCP server-defined URL for the address.
+     */
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    @Override
+    public void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        // Do nothing, API doesn't support with an update and/or patch method.
+    }
+
+    @Override
+    public void copyFrom(Address model) {
+        setName(model.getName());
+        setDescription(model.getDescription());
+        setAddress(model.getAddress());
+        setPrefixLength(model.getPrefixLength());
+        setAddressType(model.getAddressType());
+        setPurpose(model.getPurpose());
+
+        String subNetworkLink = model.getSubnetwork();
+        if (subNetworkLink != null) {
+            String sunbNetworkName = model.getSubnetwork().substring(model.getSubnetwork().lastIndexOf('/') + 1);
+
+            if (!"default".equals(sunbNetworkName)) {
+                setSubnetwork(findById(SubnetworkResource.class, sunbNetworkName));
+            }
```

Comment:
Should directly set `setSubnetwork(model.getSubnetwork())` as the getter of `self-link` of SubnetworkResource has the `@Id` annotation on it and not the name.

Reply:
Done.